test/test_atomic_open.py

```diff
@@ -0,0 +1,107 @@
+from concurrent.futures import ThreadPoolExecutor
+from pathlib import Path
+from rohmu.atomic_open import atomic_open_for_writing_bytes, atomic_open_for_writing_text, atomic_opener
+from threading import Event
+
+import os
+import pytest
+import time
+
+
+def test_file_is_atomically_created_only_after_function_execution_is_over(tmp_path: Path):
+    data_block = "x" * 100_000_000
+    # manually tested with block_count of 100 but it seems overkill to do it every time and it can hog
+    # the testing infra quite a bit.
+    # block_count = 100
+    block_count = 1
+    size = len(data_block)
+
+    output_file = tmp_path / "something"
+    should_stop = Event()
+
+    def check_not_created():
+        while not should_stop.is_set():
+            assert os.listdir(output_file.parent) == []
+            assert not output_file.exists()
+            time.sleep(0.1)
+
+    executor = ThreadPoolExecutor(max_workers=1)
+    future = executor.submit(check_not_created)
+
+    finished_writing_file_at = []
+    wait_after_write_seconds = 0.5
+
+    try:
+        assert not output_file.exists()
+        tmpfile_inode = []
+        with atomic_opener(output_file, _inode_spy=tmpfile_inode.append, mode="w") as f:
+            for unused_counter in range(block_count):
+                f.write(data_block)
+            time.sleep(3)
+            finished_writing_file_at.append(time.monotonic())
+            should_stop.set()
+            # we make sure not to trigger a race condition with check_not_created
+            time.sleep(wait_after_write_seconds)
```

Comment:
This feels flaky, looks like waiting for `future.result()` would achieve the same result.